engine/src/mac-av-player.mm

```diff
@@ -216,6 +219,21 @@ - (void)updateCurrentFrame
     m_av_player -> DoUpdateCurrentFrame(m_av_player);
 }
 
+- (void)handleCurrentTimeChanged
+{
+    if (!is_processing)
+    {
+        is_processing = YES;
+        [self performSelectorOnMainThread:@selector(doHandleCurrentTimeChanged) withObject:nil waitUntilDone:NO];
+    }
+}
+
+- (void) doHandleCurrentTimeChanged
+{
+    m_av_player->HandleCurrentTimeChanged();
+    self -> is_processing = NO;
```

Comment:
Set the flag *before* HandleCurrentTimeChanged is called. HandleCurrentTimeChanged does engine stuff, which might call wait, which might mean there's another update which occurs whilst it is processing - *after* the engine code has got the current time.